### ISSUE TITLE
Internal source file reference changes on each step

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#12438]
Usually when stepping, the viewport doesn't shift unless it needs to, like when the current line runs off the bottom. But when stepping through some scripts, it shifts on every step. I think this is for scripts that vscode doesn't have on disk (that had their source provided by the debug adapter) but I don't know why that would make a difference.

Good:
![sep-22-2016 12-20-15](https://cloud.githubusercontent.com/assets/323878/18744600/f95fa098-80be-11e6-8cca-184bbf659ac1.gif)

Bad:
![sep-22-2016 12-19-27](https://cloud.githubusercontent.com/assets/323878/18744602/fb5bebb8-80be-11e6-8ba9-da36a88d5d2f.gif)

**Comment by isidorn** [comment]
This is a blast from the past. I remember I was looking into this, but forgot what is the root cause - setting to september to investigate.

**Comment by isidorn** [comment]
This is happening because the reference of those source files changed between steps. Namely the debug adapter sends a brand new source reference between steps. Due to that on the vscode side we think those files are different and we do this 'bad stepping'.
Assigning to @weinand to investigate if node adapter could preserve the reference between steps for internal modules since we use those as an identifier on the vsocde side

**Comment by weinand** [comment]
is this with node-debug2?

**Comment by isidorn** [comment]
I see it with node-debug.
If it is also happening with node-debug2 @roblourens can comment

**Comment by roblourens** [comment]
Oh that's right, I'm surprised I never noticed that before...

**Comment by bpasero** [comment]
+1, makes it super hard to follow steps in an internal node module.

**Comment by isidorn** [comment]
This has a more serious reprocution of #13753, due to that assigning to October

**Comment by roblourens** [comment]
Fixed in node-debug2

**Comment by isidorn** [comment]
It would be awesome if we fix this also in node-debug as it makes the debug experience in internal modules quite poor.
